code/harry_potter_store/src/store.py

```diff
@@ -0,0 +1,33 @@
+class Store:
+    book_types = ["1", "2", "3", "4", "5"]
+
+    def __init__(self, book_price=8):
+        self.book_price = book_price
+        pass
+
+    def _clean(self, to_buy_books: dict) -> dict:
+        return dict(filter(lambda item: item[1] > 0 and item[0] in Store.book_types, to_buy_books.items()))
+
+    def _decrement(self, to_buy_books: dict) -> dict:
+        return self._clean(dict(map(lambda i: [i[0], i[1] - 1], to_buy_books.items())))
```

Comment:
While it works as expected, don't you think the simple old-style for-loop would be much more readable here?

Reply:
..or even dict comprehension should work as well, like `{book: amount-1 for book, amount in to_buy_books.items()}` or something similar